pallets/identity/src/lib.rs

```diff
@@ -542,16 +542,8 @@ decl_module! {
             expiry: Option<T::Moment>
         ) -> DispatchResult {
             let sender_key = AccountKey::try_from(ensure_signed(origin)?.encode())?;
-            let from_did =  match Self::current_did() {
-                Some(x) => x,
-                None => {
-                    if let Some(did) = Self::get_identity(&sender_key) {
-                        did
-                    } else {
-                        return Err(Error::<T>::NoDIDFound.into());
-                    }
-                }
-            };
+            let from_did = Context::current_identity_or::<Self>(&sender_key)
+                .ok_or_else(|| DispatchError::from(Error::<T>::NoDIDFound))?;
```

Comment:
Will it make sense to add a function in `Context` that returns did if it exists or else it returns `DispatchError::from(Error::<T>::NoDIDFound)`? that way, we can reduce all these calls to 
`let from_did = Context::current_identity_or_error::<Self>(&sender_key)?;`

Reply:
Done.